File: Algorithms and Data Structure/Features/BinarySearch.cpp
//recursion
int BinarySearch(int arr[], int l, int r, int x){
    if (l <= r) {
        int mid = l + (r - l) / 2;

        if (arr[mid] == x)
            return mid;
        if (arr[mid] < x)
            return BinarySearch(arr, mid + 1, r, x);
        else
            return BinarySearch(arr, l, mid - 1, x);
    }
    return -1;
}

//cicle
int BinarySearch(int arr[], int l, int r, int x){
    while (l <= r){
        int mid = l + (r - l) / 2;
        
        if (arr[mid] == x)
            return mid;
        
        if (arr[mid] < x)
            l = mid + 1;
        else
            r = mid - 1;
    }
    return -1;
}

File: Custom/python/algorithms/searches/binary-search.py
def binarysearch(arr, x):
    left, right = 0, len(arr)-1

    while left <= right:
        mid = left + (right - left) // 2

        if x == arr[mid]:
            return mid
        
        if x < arr[mid]:
            right = mid - 1
        else:
            left = mid
    
    return -1
